Chaincase.UI/ViewModels/StatusViewModel.cs

```diff
@@ -0,0 +1,343 @@
+                        default:
+                            return 0.3;
+                    }
+                })
+                .ToProperty(this, x => x.ProgressPercent);
+
+            IDisposable walletCheckingInterval = null;
+            Observable.FromEventPattern<bool>(typeof(Wallet), nameof(Wallet.InitializingChanged))
+                .ObserveOn(RxApp.MainThreadScheduler)
+                .Subscribe(x =>
+                {
+                    if (x.EventArgs)
+                    {
+                        TryAddStatus(StatusType.WalletLoading);
+
+                        if (walletCheckingInterval is null)
+                        {
+                            walletCheckingInterval = Observable.Interval(TimeSpan.FromSeconds(1))
+                               .ObserveOn(RxApp.MainThreadScheduler)
+                               .Subscribe(_ =>
+                               {
+                                   var wallet = Global.Wallet;
+                                   if (wallet is { })
+                                   {
+                                       var startingHeight = SmartHeader.GetStartingHeader(wallet.Network).Height;
+                                       if (wallet.LastProcessedFilter?.Header?.Height is uint lastProcessedFilterHeight
+                                            && lastProcessedFilterHeight > startingHeight
+                                            && Global.BitcoinStore?.SmartHeaderChain?.TipHeight is uint tipHeight
+                                            && tipHeight > startingHeight)
+                                       {
+                                           var allFilters = tipHeight - startingHeight;
+                                           var processedFilters = lastProcessedFilterHeight - startingHeight;
+                                           var perc = allFilters == 0 ?
+                                                100
+                                                : ((decimal)processedFilters / allFilters * 100);
+                                           TryAddStatus(StatusType.WalletProcessingFilters, (ushort)perc);
+                                       }
+
+                                       var txProcessor = wallet.TransactionProcessor;
+                                       if (txProcessor is { })
+                                       {
+                                           var perc = txProcessor.QueuedTxCount == 0 ?
+                                                100
+                                                : ((decimal)txProcessor.QueuedProcessedTxCount / txProcessor.QueuedTxCount * 100);
+                                           TryAddStatus(StatusType.WalletProcessingTransactions, (ushort)perc);
+                                       }
+                                   }
+                               }).DisposeWith(Disposables);
+                        }
```

Comment:
I know you didn't author it so motherfuck this confusing ugly code. Reactiv-ating this would be a dream

Reply:
Yeah its a bit spaghetti